### PR TITLE
[KYUUBI #3141] Trino engine etcd support

### PR DESCRIPTION
**Pull request by hddong**
### _Why are the changes needed?_

Resolved conflict jar for trino engine used etcd

### _How was this patch tested?_
- [X] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by hddong** [comment]
cc @yaooqinn @pan3793

File: externals/kyuubi-trino-engine/src/test/scala/org/apache/kyuubi/engine/trino/ShareLevelTrinoEngineTests.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.kyuubi.engine.ShareLevel
+import org.apache.kyuubi.engine.ShareLevel.ShareLevel
+import org.apache.kyuubi.operation.HiveJDBCTestHelper
+import org.apache.kyuubi.service.ServiceState
+
+/**
+ * This suite is to test some behavior with Trino engine in different share level.
+ * e.g. cleanup discovery service before stop.
+ */
+trait ShareLevelTrinoEngineTests
```
Comment (yaooqinn) [inline externals/kyuubi-trino-engine/src/test/scala/org/apache/kyuubi/engine/trino/ShareLevelTrinoEngineTests.scala:31]: we don't need to copy tests from other modules

**Comment by yaooqinn** [comment]
ha behavior should be the same across engine types, so the tests here are unnecessary to me

File: externals/kyuubi-trino-engine/src/test/scala/org/apache/kyuubi/engine/trino/ShareLevelTrinoEngineTests.scala (hunk 1 above)
Comment (hddong) [reply externals/kyuubi-trino-engine/src/test/scala/org/apache/kyuubi/engine/trino/ShareLevelTrinoEngineTests.scala]: > we don't need to copy tests from other modules

Get it, removed the tests.

**Review by yaooqinn — APPROVED** [review]

**Comment by hddong** [comment]
@pan3793 As discussed, used explicit declaration instead exclusion.
cc @yaooqinn

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**